src/server/pps/server/pipeline_controller.go

```diff
@@ -555,23 +556,69 @@ func (step *pcStep) scaleUpPipeline(ctx context.Context) (retErr error) {
 		tracing.FinishAnySpan(span)
 	}()
 
-	// compute target pipeline parallelism
-	parallelism := uint64(1)
-	if step.pipelineInfo.Details.ParallelismSpec != nil {
-		parallelism = step.pipelineInfo.Details.ParallelismSpec.Constant
+	// Compute maximum parallelism
+	maxScale := int32(1)
+	if step.pipelineInfo.Details.ParallelismSpec != nil && step.pipelineInfo.Details.ParallelismSpec.Constant > 0 {
+		maxScale = int32(step.pipelineInfo.Details.ParallelismSpec.Constant)
 	}
 
 	// update pipeline RC
-	return step.updateRC(ctx, func(rc *v1.ReplicationController) {
-		if rc.Spec.Replicas != nil && *step.rc.Spec.Replicas > 0 {
-			return // prior attempt succeeded
-		}
-		rc.Spec.Replicas = new(int32)
-		if step.pipelineInfo.Details.Autoscaling {
-			*rc.Spec.Replicas = 1
-		} else {
-			*rc.Spec.Replicas = int32(parallelism)
+	return step.updateRC(ctx, func(rc *v1.ReplicationController) bool {
+		var curScale int32
+		if rc.Spec.Replicas != nil && *rc.Spec.Replicas > 0 {
+			curScale = *rc.Spec.Replicas
+		}
+		targetScale := func() int32 {
+			if !step.pipelineInfo.Details.Autoscaling {
+				return maxScale // don't bother if Autoscaling is off
+			}
+			if curScale == 0 {
+				return 1 // make one pod to be the worker master & calculate tasks
+			}
+
+			// Master is scheduled; see if tasks have been calculated
+			taskService := step.pc.env.TaskService
+			nTasks64, _, err := taskService.TaskCount(ctx, driver.TaskNamespace(step.pipelineInfo))
```

Comment:
What does nTasks represent when queried? Is it number of Datums to process? When does this value get updated - as tasks are being completed or at the end of a job?

Reply:
`nTasks` is the number of datums in the job total. IIUC, it's computed once, by the worker master, after the Job is started